build.sbt

```diff
@@ -325,9 +325,21 @@ lazy val docs = project("docs")
   .settings(
     name := "akka-http-docs",
     resolvers += Resolver.jcenterRepo,
-    // In docs adding an unused variable can be helpful, for example
-    // to show its type
-    scalacOptions += "-Xlint:-unused",
+    scalacOptions ++= Seq(
+      // Make sure we don't accidentally keep documenting deprecated calls
+      "-Xfatal-warnings",
+      // In docs adding an unused variable can be helpful, for example
+      // to show its type
+      "-Xlint:-unused",
+      // We use this for `complete`
```

Comment:
Good point, added a comment about that to https://github.com/akka/akka-http/issues/2738#issuecomment-621109993.